index.src.html

```diff
@@ -1146,6 +1146,48 @@ <h3 id="gc">Garbage Collection</h3>
   removed from the <a>report buffer</a> of any <a>reporting observer</a>.
 </section>
 
+<section>
+  <h2 id="deployment">Deployment Considerations</h2>
+
+  <h3 id="custom-metadata">Custom Metadata</h3>
+
+  A server might want to include additional metadata in reports that are
+  generated for their origin.  This can be accomplished by encoding the extra
+  metadata in the <a for="ReportTo">`url`</a> of any <a
+    for="ReportTo">`endpoints`</a> in the <a>`Report-To`</a> response headers
+  for the origin — for example, in the URL path or query parameters.
+
+  <pre>
+    <a>Report-To</a>: { "<a for="ReportTo">group</a>": "csp",
+                 "<a for="ReportTo">max-age</a>": 10886400,
+                 "<a>endpoints</a>": [
+                   { "<a for="ReportTo">url</a>": "https://example.com/reports?nonce=e897932f" }
+                 ] }
+  </pre>
+
+  Since the instructions in a <a>`Report-To`</a> header will be used for future
+  requests to the same origin, the server SHOULD NOT use this mechanism to
+  encode metadata that is only valid for the current request.  The metadata MUST
+  be valid for all requests to the same origin from the same user.
+
+  <h3 id="spam-mitigation">Spam Mitigation</h3>
+
+  One potential use of [[#custom-metadata]] is to help prevent spam — report
+  uploads that don't correspond to a real request made by a real user.  For
+  instance, when constructing the <a>`Report-To`</a> for a response, the server
+  could create a nonce whose value depends on the origin of the request, and the
+  public IP address of the client.  The server would then embed this nonce into
+  the <a for="ReportTo">`url`</a> values of the header.
```

Comment:
What happens when I migrate to a different network and a report is triggered? E.g. I switch from coffee shop wifi to my phone, or move from work to home wifi?

Reply:
Chrome will throw those reports away, on the assumption that the user might have made requests on one network that they wouldn't have made on another, and doesn't want that information to leak across the network boundary.

If it didn't do that, though, the server would throw away the requests that were uploaded from a different IP than was used to make the original request.  You can leave that out of the nonce calculation, but then a spammer could upload false reports from a huge number of clients using a single upload URL.

Can add some text about there being a trade-off — if you include more fields in the nonce, you're better protected against false uploads, but you might also miss out on some legitimate ones; and vice versa.

Reply:
Hmm, this has interesting implications..

a) Presumably the network switch logic wouldn't apply to all report types? For example, it's not clear to me why we'd clear deprecation reports on a network switch.

b) Drop on network switch has interesting implications for NEL.. This means that we're blind to flaky networks or networks that block user traffic for some reason. That said, this is a discussion for NEL so we can defer that here.

Reply:
> a) Presumably the network switch logic wouldn't apply to all report types? For example, it's not clear to me why we'd clear deprecation reports on a network switch.

That would add to the boilerplate when defining a new event type.  (I'm not against it, though — just walking through the ramifications!)  Right now you'd have:

- format of the report `body`
- observable from JavaScript?
- clear cache on network switch?

Reply:
> That would add to the boilerplate when defining a new event type.

Do you mean specification boilerplate or user-visible boilerplate?

Reply:
Spec

Reply:
OK. In that case, IMO that's fine